### PR TITLE
Trace TVar in IOSim & IOSimPOR

### PR DESCRIPTION
**Pull request by coot**
Fixes: #3601
Requires: #3603

- io-classes: MonadTraceSTM

### CODE REVIEW

**Review by bolt12 — APPROVED** [review]
Looks nice :D

File: io-classes/src/Control/Monad/Class/MonadSTM.hs [hunk 1]
```diff
@@ -238,6 +243,130 @@ class MonadSTM m
   default labelTBQueueIO :: TBQueue m a -> String -> m ()
   labelTBQueueIO = \v l -> atomically (labelTBQueue v l)
 
+
+-- | This type class is indented for 'io-sim', where we one might want to
+-- access 'TVar' in the underlying 'ST' monad.
```
[the remark this answers is not in the document]
Comment (coot) [reply io-classes/src/Control/Monad/Class/MonadSTM.hs:248]: actually no, they are accessible in `ST` monad.

File: ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/MockEnvironment.hs [hunk 2]
```diff
@@ -242,15 +243,20 @@ mockPeerSelectionActions tracer
                                }) <- adjacency
                    ]
     targetsVar <- playTimedScript (contramap TraceEnvSetTargets tracer) targets
-    peerConns  <- newTVarIO Map.empty
+    peerConns  <- atomically $ do
```
[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/MockEnvironment.hs:246]: to be precise, every time when a `TVar` is committed:
* it was written
* the transaction is committed

[the remark this answers is not in the document]
Comment (bolt12) [reply ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/MockEnvironment.hs:246]: Maybe worth making a comment making that explicit. Either here or in the interface file

[the remark this answers is not in the document]
Comment (coot) [reply ouroboros-network/test/Test/Ouroboros/Network/PeerSelection/MockEnvironment.hs:246]: Done; I also added a `noop` haddock to the `IO` instance.

**Pull request merged**